drv/cpu-seq-api/src/lib.rs

```diff
@@ -31,6 +32,22 @@ pub enum SeqError {
     ServerRestarted,
 }
 
+#[derive(Copy, Clone, Debug, FromPrimitive, Eq, PartialEq, AsBytes, Count)]
+#[repr(u8)]
+pub enum StateChangeReason {
+    /// TThe system has just received power, so the sequencer has booted the
+    /// host CPU.
+    InitialPowerOn = 1,
+    /// A power state change was requested by the control plane.
+    ControlPlane,
+    /// The host OS requested that the system power off without rebooting.
+    HostPowerOff,
+    /// The host OS requested that the system reboot.
+    HostReboot,
+    /// The system powered off because a component has overheated.
+    Overheat,
+}
```

Comment:
I'm very open to suggestions about the naming of these variants, if anyone dislikes the ones I came up with...